### PR TITLE
Added option to open folder

### PR DESCRIPTION
**Pull request by Jannify**

### CODE REVIEW

**Comment by Jannify** [comment]
Fixed a bug were automatically opening the last selected TypeTreeView resulted in displaying wrong files.

File: TypeTreeDiffGUI/UnityVersionComparer.cs [hunk 1]
```diff
@@ -0,0 +1,366 @@
...
+	    Alpha = 0,
+	    Beta,
+	    China,
+	    Final,
+	    Patch,
+	    Experimental,
+
+	    MaxValue = Experimental,
+    }
+}
```
Comment (ds5678) [inline TypeTreeDiffGUI/UnityVersionComparer.cs]: Use the nuget package

File: .gitignore [hunk 2]
```diff
@@ -1,8 +1,409 @@
...
+# Ignore only specific Rider files/directories
+# For syntax see: https://stackoverflow.com/a/14422549/12124
+/.idea/*
+!/.idea/.idea.*
+/.idea/.idea.*/*
+!/.idea/.idea.*/.idea
+/.idea/.idea.*/.idea/*
+!/.idea/.idea.*/.idea/runConfigurations
+
+
```
Comment (ds5678) [inline .gitignore]: This wasn't necessary

File: TypeTreeDiffGUI/DropArea.xaml.cs [hunk 3]
```diff
@@ -55,8 +57,16 @@ private void OnDropped(object sender, DragEventArgs e)
 		{
 			if (e.Data.GetDataPresent(DataFormats.FileDrop))
 			{
-				string[] files = (string[])e.Data.GetData(DataFormats.FileDrop);
-				EventFileDropped?.Invoke(files[0]);
+				string filePath = ((string[])e.Data.GetData(DataFormats.FileDrop))?[0];
```
Comment (ds5678) [inline TypeTreeDiffGUI/DropArea.xaml.cs]: Check that the length is exactly one and throw an exception otherwise

**Comment by ds5678** [comment]
Sorry, I didn't see this until now

File: TypeTreeDiffGUI/UnityVersionComparer.cs (hunk 1 above)
Comment (Jannify) [reply TypeTreeDiffGUI/UnityVersionComparer.cs]: There is a package for that 👀 What is it called?

File: .gitignore (hunk 2 above)
Comment (Jannify) [reply .gitignore]: Should I just add the (for me necessary) Rider part then?

File: TypeTreeDiffGUI/DropArea.xaml.cs [hunk 4]
```diff
@@ -55,8 +57,26 @@ private void OnDropped(object sender, DragEventArgs e)
...
-				EventFileDropped?.Invoke(files[0]);
+				string[] filePaths = ((string[])e.Data.GetData(DataFormats.FileDrop));
+
+				if (filePaths == null || filePaths.Length == 0)
+				{
+					throw new ArgumentException("No files or folders were dropped.");
+				}
+				if (filePaths.Length > 1)
+				{
+					throw new ArgumentException("Multiple files or folders were dropped but only one is supported.");
```
Comment (ds5678) [inline TypeTreeDiffGUI/DropArea.xaml.cs]: I don't think ArgumentException is appropriate for these. Just Exception should be fine. Other than that, this PR should be good.

**Pull request merged**